### PR TITLE
Teapot Implementations & Fixes

### LINKED ISSUE #2051: [Bug] Dust Song Pot trust level cant increase 尘歌壶信任等阶无法提升
game 3.4 游戏版本
gc 1.4.6 服务器版本

The trust level and trust value of Dusty Song Pot cannot be increased

It is invalid to use give to obtain the trust value and make the decoration for the first time

尘歌壶信任等阶和信任值无法提升
使用give获取信任值以及首次制作摆设均无效

### PR DESCRIPTION
**Pull request by NotThorny**
## Description

Fixes and additional implementations for the teapot system.

- [x] Unlocking realms from their respective items
- [x] Switching between realms
- [x] Exiting the teapot to where you originally entered
- [x] Claim home level rewards
- [x] Fixes Grass main house bricking accounts
- [x] Fixes Grass realm due to house fix (Grass realm only: main house needs to be placed manually)
- [x] Adds gaining trust (realm exp) from crafting and with `/give`
- [x] Make trust values show in djinn menu

Example of some of the functions:
https://streamable.com/nrp03d

## Issues fixed by this PR

Fixes #2051 

## Type of changes

- [x] Bug fix
- [x] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseUnlockHomeModule.java [hunk 1]
```diff
@@ -14,6 +14,13 @@ public ItemUseUnlockHomeModule(String[] useParam) {
 
     @Override
     public boolean useItem(UseItemParams params) {
-        return false;
+        return true;
+    }
+
+    @Override
+    public boolean postUseItem(UseItemParams params){
+
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseUnlockHomeModule.java]: remove whitespace

File: src/main/java/emu/grasscutter/server/packet/recv/HandlerBackMyWorldReq.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import emu.grasscutter.server.event.player.PlayerTeleportEvent.TeleportType;
+import emu.grasscutter.server.game.GameSession;
+import emu.grasscutter.server.packet.send.PacketBackMyWorldRsp;
+
+@Opcodes(PacketOpcodes.BackMyWorldReq)
+public class HandlerBackMyWorldReq extends PacketHandler {
+	
+	@Override
+	public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
+
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/server/packet/recv/HandlerBackMyWorldReq.java]: whitespace

File: src/main/java/emu/grasscutter/server/packet/recv/HandlerHomeChangeModuleReq.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import emu.grasscutter.net.packet.PacketHandler;
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.HomeChangeModuleReqOuterClass;
+import emu.grasscutter.server.event.player.PlayerTeleportEvent.TeleportType;
+import emu.grasscutter.server.game.GameSession;
+import emu.grasscutter.server.packet.send.PacketHomeChangeModuleRsp;
+import emu.grasscutter.server.packet.send.PacketHomeComfortInfoNotify;
+import emu.grasscutter.server.packet.send.PacketPlayerHomeCompInfoNotify;
+import emu.grasscutter.utils.Position;
+
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/server/packet/recv/HandlerHomeChangeModuleReq.java:14]: spacing

File: src/main/java/emu/grasscutter/server/packet/send/PacketPlayerEnterSceneNotify.java [hunk 4]
```diff
@@ -43,6 +43,14 @@ public PacketPlayerEnterSceneNotify(Player player, EnterType type, EnterReason r
     public PacketPlayerEnterSceneNotify(Player player, Player target, EnterType type, EnterReason reason, int newScene, Position newPos) {
         super(PacketOpcodes.PlayerEnterSceneNotify);
 
+        // Set previous position
+        if(!(newScene == 3)){ // Hardcoded for now else weird positions will occur
+            // Don't update position within same scene or teapot
+        }else{
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/server/packet/send/PacketPlayerEnterSceneNotify.java]: Fix spacing.
`if (`
`} else {`

**Review by KingRainbow44 — DISMISSED** [review]
i feel rlly bad if whitespace bot goes in automatically and fixes whitespace lol

File: src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseUnlockHomeModule.java (hunk 1 above)
Comment (NotThorny) [reply src/main/java/emu/grasscutter/game/props/ItemUseAction/ItemUseUnlockHomeModule.java]: Whitespace and formatting should be fixed now.

**Review by GanyusLeftHorn — APPROVED** [review]

**Pull request merged**